ha/core/action_handler/action_factory.py

```diff
@@ -41,9 +39,13 @@ def get_action_handler(event: HealthEvent, action: list) -> ActionHandler:
             ActionHandler: return Specific object of ActionHandler like NodeFruPSUActionHandler for node:fru:psu.
         """
         try:
-            if event.resource_type in EVENT_ACTION_HANDLERS_MAPPING:
+            if len(actions) == 0 or \
+                    len(actions) == 1 and \
```

Comment:
Why this code has been changed? Did we see any issues?
Ideally we should not get here with actions length being zero.
Also, why we want to call the DefaultActionHandler() with length 1 with "publish" action? That way the resource specific action handlers with just "publish" won't be called.

Reply:
Yes previous logic had some issue
if we have nodeactionhandler but ha rule is not added and one component subscribe to node then we need default method 
we need specific handler only if we have multiple action or ha action in actions list